### ISSUE TITLE
Would like to see more explanation

### ISSUE THREAD

**Issue by krumiaa** [Coac/neat-tfjs#1]
This looks great, any chance you will be adding further documentation and examples? Specifically, how could this be implimented to solve a simple reinforcement learning task?

**Comment by Coac** [comment]
Hello, thanks,
I suggest you read the paper first. In the code, I used the same terminology as described in it.
For now, it's not working very well, so if you want to contribute to the repository you are welcome ;)
For neat, it's more a black-box optimization algorithm than a real RL algo. You cant apply directly to an easy task, generating the network and using it to computes the actions.
But you can also try to use neat to generate a neural net and then apply any RL algorithm for some updates and use the average return of your agent like the fitness. However, I think it will take too much time, but for an easy task, why not, you can try if you have time.

**Comment by krumiaa** [comment]
Appreciate your quick response I would love to contribute, but at this point still trying to figure out the nuts and bolts of NEAT, not sure how you even got this working as from my understanding tensorflow does not allow dynamic computational graphs, eg changing the size and shape of the NN in runtime. Maybe you found a workaround? Anyway I'm intrigued by the possibilities with RL - seemingly Uber labs has recently demonstrated that these genetic algorithms can outperform standard RL techniques like deep Q. At leas that was my take away from the research published here https://eng.uber.com/deep-neuroevolution/

**Comment by Coac** [comment]
I am using tensorflow js, it uses eager execution, so you can create a dynamic model. You can see the code related to his here https://github.com/Coac/neat-tfjs/blob/master/neat/TFGenome.js
Yes saw the Uber article too, they used lots of computes. If you want to use neat to achieve the same goal, you will need to parallelize the algorithm otherwise I think it will be too slow.

**Comment by krumiaa** [comment]
Ok, I must have misunderstood the algorithm  because I thought the appeal of NEAT was that it could rival DQN and policy gradients with allot less time training and computing power. Wonder if I got swept up in the hype. Is there a variation you know of that would is more efficient and robust, like HyperNEAT?

**Comment by Coac** [comment]
I am not sure that neat is competitive against deep RL, but scalable neuroevolution is. But which is less sample efficient and requiring more compute power.
